src/test/scala/io/qbeast/spark/utils/QbeastSparkTxnTest.scala

```diff
@@ -91,14 +97,20 @@ class QbeastSparkTxnTest extends QbeastIntegrationTestSpec {
         .option(QbeastOptions.COLUMNS_TO_INDEX, "id")
         .option(QbeastOptions.CUBE_SIZE, 1)
         .option(QbeastOptions.TXN_APP_ID, "test")
-        .option(QbeastOptions.TXN_VERSION, "1")
+        .option(QbeastOptions.TXN_VERSION, "2")
```

Comment:
A quick question, looking at this test scenario, it is not clear to me how we know that txn version 1 is a version already committed. I'm making the assumption that we are changing to 2 because with 1 it fails, but expanding the lines and looking at the case, I'm not sure why that holds. Please help me understand it.

Reply:
@fpj Step 2 specifies when a `txnVersion` is ignored, and only the latest version for a given `appId` is stored in the **snapshot**.

> For example, the [Delta Sink for Apache Spark's Structured Streaming](https://github.com/delta-io/delta/blob/master/core/src/main/scala/org/apache/spark/sql/delta/sources/DeltaSink.scala) ensures exactly-once semantics when writing a stream into a table using the following process:
> 
> 1. Record in a write-ahead-log the data that will be written, along with a monotonically increasing identifier for this batch.
> 2. Check the current version of the transaction with appId = streamId in the target table. **If this value is greater than or equal to the batch being written**, then this data has already been added to the table and processing can skip to the next batch.
> 3. Write the data optimistically into the table.
> 4. Attempt to commit the transaction containing both the addition of the data written out and an updated appId version pair.
> 
> The semantics of the application-specific version are left up to the external system. **Delta only ensures that the latest version for a given appId is available in the table snapshot**. The Delta transaction protocol does not, for example, assume monotonicity of the version and it would be valid for the version to decrease, possibly representing a "rollback" of an earlier transaction.

Reply:
Thanks, @Jiaweihu08, but I think you are responding to my other question. In this thread, I'm asking for a clarification on the test case, could you have another look, please?

Reply:
Ok, so where are we validating that a latter attempt is being ignored? My guess is that the `data.write` in lines 108-113 should be ignored, and if it isn't, then the count in line 114 is going to be incorrect and the test case will fail, is that it?

Reply:
Yes, as well as the one from 102-107.